src/main/java/core/basesyntax/BinaryString.java

```diff
@@ -7,6 +7,18 @@
      * а возвращает String с представлением этого числа в двоичном виде.
      */
     public String toBinaryString(int value) {
-        return null;
+        if (value == 0) {
+            return "0";
+        }
+        StringBuilder sb = new StringBuilder();
+        while (value != 0) {
+            if ((value & 1) == 1) {
```

Comment:
do we need this `if` statement?
hint:
what result can the expression return: `value & 1`?

Reply:
Omg, you are absolutely correct we no need it there! It is much much better now!